internal/pkg/cli/selector.go

```diff
@@ -0,0 +1,189 @@
+package cli
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/color"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/term/log"
+)
+
+// selector prompts users to select an app, environments, or service.
+// We can use this instead of repeating our own selectors in our own
+// commands.
+type selector struct {
+	prompt prompter
+}
+
+type selectWorkspaceServiceRequest struct {
+	Prompt string          // instructs users on what to select
+	Help   string          // provides help text on what users are selecting
+	Lister wsServiceLister // lists local services
+}
+
+type selectServiceRequest struct {
+	Prompt string        // instructs users on what to select
+	Help   string        // provides help text on what users are selecting
+	App    string        // the app whose services you'd like to list
+	Lister serviceLister // lists all services for an app
+}
+
+type selectEnvRequest struct {
+	Prompt string            // instructs users on what to select
+	Help   string            // provides help text on what users are selecting
+	App    string            // the app whose envs you'd like to list
+	Lister environmentLister // lists all envs in an app
+}
+
+type selectAppRequest struct {
+	Prompt string            // instructs users on what to select
+	Help   string            // provides help text on what users are selecting
+	Lister applicationLister // lists all apps in an account and region
+}
+
+// WorkspaceApp fetches all services in the workspace and then prompts the
+// user select one.
+func (s *selector) WorkspaceService(req *selectWorkspaceServiceRequest) (string, error) {
```

Comment:
should these methods be named `Select*`? like `SelectWorkspace` or `SelectService`

Reply:
Yea, let's do that.